src/components/landing/pages/history/history.tsx

```diff
@@ -75,32 +122,45 @@ export const History: React.FC = () => {
     })
   }
 
-  const tags = historyEntries.map(entry => entry.tags)
-    .reduce((a, b) => ([...a, ...b]), [])
-    .filter((value, index, array) => {
-      if (index === 0) {
-        return true
-      }
-      return (value !== array[index - 1])
-    })
-  const entriesToShow = sortAndFilterEntries(historyEntries, viewState)
+  const tags = useMemo<string[]>(() => {
+    return mergeEntryArrays(localHistoryEntries, remoteHistoryEntries).map(entry => entry.tags)
```

Comment:
if we have entries to show already shouldn't we maybe use that here?

Reply:
No, because entriesToShow is already filtered. For the tags we need the unfiltered data.